src/AppEndToEndAlternateBallotStyle.test.tsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react'
+import { fireEvent, render } from 'react-testing-library'
+
+import electionSample from './data/electionSample.json'
+
+import App, { electionKey, mergeWithDefaults } from './App'
+import { CandidateContest, Election, YesNoContest } from './config/types'
+
+const electionSampleAsString = JSON.stringify(
+  mergeWithDefaults(electionSample as Election)
+)
+
+const presidentContest = electionSample.contests.find(
+  c => c.title === 'President and Vice-President' && c.seats === 1
+) as CandidateContest
+
+const measure102Contest = electionSample.contests.find(
+  c =>
+    c.title === 'Measure 102: Vehicle Abatement Program' && c.type === 'yesno'
+) as YesNoContest
+
+beforeEach(() => {
+  window.localStorage.clear()
+  window.location.href = '/'
+})
+
+it(`basic end-to-end flow`, async () => {
+  /* tslint:disable-next-line */
+  const eventListenerCallbacksDictionary: any = {}
+  window.addEventListener = jest.fn((event, cb) => {
+    eventListenerCallbacksDictionary[event] = cb
+  })
+  window.print = jest.fn(() => {
+    eventListenerCallbacksDictionary.afterprint()
+  })
+
+  window.localStorage.setItem(electionKey, electionSampleAsString)
+  const { container, getByText, getByTestId, queryByText } = render(<App />)
+
+  getByText('Scan Your Activation Code')
+
+  fireEvent.change(getByTestId('activation-code'), {
+    // the alternate activation code that picks the other ballot style
+    target: { value: 'VX.precinct-21.5R' },
+  })
+
+  // TODO: replace next line with "Enter" keyDown on activation code input
+  fireEvent.click(getByText('Submit'))
+
+  // Get Started Page
+  fireEvent.click(getByText('Get Started'))
+  expect(container.firstChild).toMatchSnapshot()
+
+  // make sure there are as many questions as we expect
+  getByText('1 of 11')
```

Comment:
Can we end the test right here? 

Is there anything after this which is specific to this test?